lib/gc/Transforms/IterativeTilingAndFusion.cpp

```diff
@@ -68,17 +69,21 @@ getClosestInsertSliceOfResult(OpResult result) {
       sliceOp =
           dyn_cast<OffsetSizeAndStrideOpInterface>(useOfResult.getOwner());
     } else if (auto yieldOp = dyn_cast<scf::YieldOp>(useOfResult.getOwner())) {
-      if (auto loop = dyn_cast<LoopLikeOpInterface>(yieldOp->getParentOp()))
+      if (isa<LoopLikeOpInterface, scf::IfOp>(yieldOp->getParentOp())) {
         return getClosestInsertSliceOfResult(
-            loop->getResult(useOfResult.getOperandNumber()));
+            yieldOp->getParentOp()->getResult(useOfResult.getOperandNumber()));
+      }
+    } else if (isa<tensor::ExpandShapeOp, tensor::CollapseShapeOp>(
```

Comment:
Better to add an `else` clause to capture unexpected cases like this new one?

Reply:
I am afraid not because no branch hit is also expected case in this context.